src/Core/Form/IdentifiableObject/Builder/FormBuilder.php

```diff
@@ -110,28 +133,45 @@ public function getForm(array $data = [], array $options = [])
      */
     public function getFormFor($id, array $data = [], array $options = [])
     {
-        $data = array_merge($this->dataProvider->getData($id), $data);
+        // Fetch initial data
+        $initialData = $this->dataProvider->getData($id);
+        // Fetch initial options
+        $initialOptions = $options;
+
+        $this->hookDispatcher->dispatchWithParameters(
+            'action' . Container::camelize($this->registry->getType($this->formType)->getBlockPrefix()) . 'FormDataProviderData',
```

Comment:
We may need a hook before as well but I'm not sure
At least this hook should be placed after L157 when `$this->optionsProvider->getOptions` has been called (same in the default method)

Reply:
Changed. Thanks :)